src/toil/fileStore.py

```diff
@@ -703,6 +707,8 @@ def readGlobalFile(self, fileStoreID, userPath=None, cache=True, mutable=False,
                         # recording it in the cache lock file) if possible.
                         if os.path.exists('/.'.join(os.path.split(cachedFileName))):
                             os.rename('/.'.join(os.path.split(cachedFileName)), cachedFileName)
+                            # If this is not true we get into trouble in our internal reference counting.
+                            assert(os.stat(cachedFileName).st_nlink == self.nlinkThreshold)
```

Comment:
assert is removed with compiling to optimised byte code (python -o producing *.pyo files). Maybe convert into an exception?

Reply:
I think this is properly an assert; if we're right in our reasoning about how the code functions (i.e. the FileStore and the FileJobStore agree about link counts), it should never be hit. It is safe to skip if you are willing to sacrifice debuggability for speed using `-o`.

Reply:
Okay, and all the other asserts?